pkg/gateway/lb/handler/grpc/handler.go

```diff
@@ -1713,81 +1713,81 @@ func (s *server) MultiUpsert(ctx context.Context, reqs *payload.Upsert_MultiRequ
 		}
 	}
 
-	insertLocs := make([]*payload.Object_Location, 0, len(insertReqs))
-	updateLocs := make([]*payload.Object_Location, 0, len(updateReqs))
-
-	var (
-		errs error
-		mu   sync.Mutex
-	)
-	eg, ectx := errgroup.New(ctx)
-	eg.Go(safety.RecoverFunc(func() error {
-		if len(updateReqs) <= 0 {
-			return nil
-		}
-
-		ectx, span := trace.StartSpan(ectx, apiName+".MultiUpsert/Go-MultiUpdate")
-		defer func() {
-			if span != nil {
-				span.End()
-			}
-		}()
-		var err error
-		loc, err := s.MultiUpdate(ectx, &payload.Update_MultiRequest{
+	switch {
+	case len(insertReqs) <= 0:
+		res, err = s.MultiUpdate(ctx, &payload.Update_MultiRequest{
 			Requests: updateReqs,
 		})
-		if err == nil {
-			updateLocs = loc.GetLocations()
-		} else {
-			mu.Lock()
-			if errs == nil {
-				errs = err
-			} else {
-				errs = errors.Wrap(errs, err.Error())
+	case len(updateReqs) <= 0:
+		res, err = s.MultiInsert(ctx, &payload.Insert_MultiRequest{
+			Requests: insertReqs,
+		})
+	default:
+		var (
+			ures, ires *payload.Object_Locations
+			errs       error
+			mu         sync.Mutex
+		)
+		eg, ectx := errgroup.New(ctx)
+		eg.Go(safety.RecoverFunc(func() (err error) {
+			ures, err = s.MultiUpdate(ectx, &payload.Update_MultiRequest{
```

Comment:
I'm not sure, but can the results be affected when calling MultiUpdate and MultiUpsert in parallel? 🤔

Reply:
I didn't call MultiUpdate and MultiUpsert in parallel.

Reply:
I am sorry. I made a mistake. :bow:
It is about calling MultiUpdate and MultiInsert in parallel as shown below.
https://github.com/vdaas/vald/blob/fafd417d39444b24dda21b5465cc2c935f29d909/pkg/gateway/lb/handler/grpc/handler.go#L1732-L1761

Reply:
I am not sure what your concern is.
However, what I am interested in is what problems you imagine would occur if MultiUpdate and MultiInsert were called in parallel.

Reply:
I understood. I read it again, so I realized that I made a mistake. sorry.
There is no problem with your implementation. I will close this comment.